Feature/ela 51 general asset upload repository public

Gavindecode opened the pull request:
## Kanopi

https://kanopi.live/app/ticket/49289

## Key Points

- **Code Cleanup**: Removed unnecessary blank lines and performed structural refactors to improve code readability and consistency across files.
- **File Structure and Dependencies**: Updated `package.json` dependencies and added resources in `readme.md`. Revised routing implementations to align with new components.
- **File and Folder Management**: Introduced popups and menu actions for adding, renaming, and deleting files and folders. Integrated new validations for folder names.
- **Search Functionality**: Added robust search features and placeholder text to improve search usability.
- **File Editing Enhancements**: Completed file renaming feature with backend validations and improved error feedback on UI.
- **Drag-and-Drop File Upload**: Integrated a file upload component supporting progress tracking. Improved user interaction details and responsiveness.
- **UI Consistency**: Expanded support for multilingual configurations, reorganized resource files, and replaced legacy components.
- **File Metadata**: Designed metadata handling and file categorization for better clarity in UI.

## Screenshots

Gavindecode committed:
Refactor Fileman storage handling and add root folder migration.

Update storage logic to use a configurable filesystem disk throughout Fileman, enhancing flexibility and maintainability. Add a migration to ensure the presence of a root folder in the database. Other updates include fixing file path generation and improving blade templates for consistent image handling.

Gavindecode committed:
Add ESLint setup and UI improvements for Fileman package

Introduce ESLint configuration with rules and ignore files to enhance code quality. Update various views to improve UI consistency and multilingual support. Add `FileService` methods for file counting and initialize front-end integration with Vue.

Gavindecode committed:
Add ESLint setup and UI improvements for Fileman package

Introduce ESLint configuration with rules and ignore files to enhance code quality. Update various views to improve UI consistency and multilingual support. Add `FileService` methods for file counting and initialize front-end integration with Vue.

Gavindecode committed:
Add ESLint setup and UI improvements for Fileman package

Introduce ESLint configuration with rules and ignore files to enhance code quality. Update various views to improve UI consistency and multilingual support. Add `FileService` methods for file counting and initialize front-end integration with Vue.

Gavindecode committed:
Add ESLint setup and UI improvements for Fileman package

Introduce ESLint configuration with rules and ignore files to enhance code quality. Update various views to improve UI consistency and multilingual support. Add `FileService` methods for file counting and initialize front-end integration with Vue.

Gavindecode committed:
Add PopupLayout and AddFolderPopup components

Introduce reusable PopupLayout and AddFolderPopup Vue components to streamline UI for modal dialogs. Updated directory and folder views to integrate these new components, added corresponding SVG icons, and adjusted related props and types. Modified JavaScript imports and structure for better component management.

Gavindecode committed:
Add file upload functionality and UI improvements

Introduced a drag-and-drop file upload component with progress tracking and integrated it into the file management interface. Enhanced the user experience with updated styles, responsiveness, and layout adjustments for file and folder components.

Gavindecode committed:
Add file management UI enhancements and file metadata support

Replaced legacy file component with a Vue-based solution for better UI consistency and interactivity. Added support for file metadata display, including type, size, and preview thumbnails. Updated configuration to include file type categorization with related colors for improved clarity.

Gavindecode committed:
Add file renaming feature with validation and UI updates

Implemented file renaming functionality with necessary validation rules in the backend, including custom file name validation logic. Updated frontend components and templates to support renaming files, improved UI for better feedback, and updated API routes for file operations.

Gavindecode committed:
Add validation for folder names and improve UI consistency

Introduced a new `ValidFolderName` rule to validate folder names, ensuring they meet specific criteria. Updated styles in `RenameFilePopup.vue` for consistent theming and added error alerts in `AddFolderPopup.vue` for better user feedback. Enhanced language file with additional keys to support new validation messages.

Gavindecode committed:
Add search functionality with file and folder resources

Introduced search services and endpoints for files and folders, including changes to FolderService and FileService. Added new API routes, resource classes, and UI updates to support search. Enhanced existing components with improved layout and styling adjustments.

Gavindecode committed:
Add search functionality with file and folder resources

Introduced search services and endpoints for files and folders, including changes to FolderService and FileService. Added new API routes, resource classes, and UI updates to support search. Enhanced existing components with improved layout and styling adjustments.

Gavindecode committed:
Add "placeholders" array with search placeholder text

Enhanced the localization file by introducing a "placeholders" section. This includes a "search" key to provide placeholder text for the file manager search functionality. This change improves usability and clarity in the UI.

Gavindecode committed:
Add folder and file management popups and menu actions

Introduced components and logic for adding, renaming, and deleting files and folders using popups and menu actions. Updated route mappings and API interactions to support these functionalities. Revised UI components to integrate the new features seamlessly with enhanced user interactions.

Gavindecode committed:
Reformat and enhance file structure and dependencies

Refactored and restructured various code files for improved readability, including configuration files and language resources. Introduced new dependencies in `package.json` and updated installation instructions in `readme.md`. Minor fixes in resource files and route implementation to ensure consistency and compatibility.

Gavindecode committed:
Remove unnecessary blank lines for cleaner code

Eliminated excessive blank lines across multiple Vue components and a service provider file to improve code readability and maintain a consistent structure. No functional changes were introduced.

Gavindecode committed:
Refactor popup buttons' styles for consistency.

Updated button styles across DeleteFolderPopup, DeleteFilePopup, RenameFilePopup, and RenameFolderPopup components to ensure uniformity. Changes include full-width buttons, rounded-full borders, adjustments to padding, and improved styling for disabled states.

Gavindecode committed:
Update layout and grid styles for consistency

Adjusted grid classes in folder and file views for uniformity. Added a minimum width to the page layout and refined modal width in FileDetail.vue for better responsiveness. These changes improve design consistency and user experience across components.

Gavindecode committed:
Update folder creation route to use API endpoint

Switched the "create-folder-url" to point to the API route for folder creation instead of the standard web route. This ensures the action aligns with API-first principles and improves consistency across the application.

michellezhu001 commented:
@Gavindecode Can you please upload screenshots or preferably video recording for this feature?

lionslair commented on package.json:
can not install

![Screenshot from 2025-02-10 08-19-08](https://github.com/user-attachments/assets/75461f0f-9267-40ab-bea9-b7645178db4d)

lionslair commented on readme.md:
````diff
@@ -12,6 +12,17 @@ Install the package via composer:
+
 Publish the migrations:
 
 ```php
````
should be `bash`. You want them to run a command

lionslair commented on src/Http/Controllers/Api/SearchController.php:
```diff
@@ -0,0 +1,26 @@
+class SearchController extends Controller
+{
+
+    public function __construct(public SearchService $searchService)
```
`$searchService` should be protected. Why did you make it public?

lionslair commented on src/Http/Controllers/FolderController.php:
```diff
@@ -44,7 +47,7 @@ public function store(FolderRequest $request, Folder $parent): RedirectResponse
         return redirect()
             ->route('fileman.folder.index', Folder::create([
-                'name' => $request->input('name'),
+                'name' => $request->name,
```
Why did you change this? Revert it

lionslair commented on src/Http/Resources/List/File.php:
```diff
@@ -0,0 +1,40 @@
+            parent::toArray($request),
+            [
+                'url' => $this->resource->getUrl(),
+                'previewUrl' => $this->resource->hasPreview()?$this->resource->getPreview():$this->resource->getUrl(),
```
Formatting. What is your issue with spaces?

lionslair committed:
Update to use pint to fix formatting

lionslair committed:
Fix PHP Syntax Styling

lionslair committed:
Import the code quality tools

lionslair committed:
Merge branch 'feature/ELA-51-general-asset-upload-repository-public' of github.com:DCODE-GROUP/fileman into feature/ELA-51-general-asset-upload-repository-public

lionslair committed:
Fix PHP Syntax Styling

lionslair committed:
cleanup

lionslair committed:
Fix the resources

lionslair committed:
Fix formatting

lionslair committed:
Fix getting storage url

lionslair committed:
Fix PHP Syntax Styling

lionslair committed:
fix type hint

lionslair committed:
Merge branch 'feature/ELA-51-general-asset-upload-repository-public' of github.com:DCODE-GROUP/fileman into feature/ELA-51-general-asset-upload-repository-public

lionslair committed:
fix type hint

lionslair committed:
Fix PHP Syntax Styling

lionslair commented on src/Services/FileService.php:
```diff
@@ -4,17 +4,21 @@
         $name = $name ?: $file->getClientOriginalName();
-        $filename = uniqid().'-'.$name;
-        $source = Storage::disk('s3')->putFileAs($path, $file, $filename);
+        $filename = uniqid().'-'.str_replace(' ', '_', $name);
```
WHY WHY WHY not just use the hash name the system generates?

lionslair committed:
fix formatting

lionslair committed:
Merge branch 'feature/ELA-51-general-asset-upload-repository-public' of github.com:DCODE-GROUP/fileman into feature/ELA-51-general-asset-upload-repository-public

lionslair reviewed (changes requested):
A lot of changes and cleanups already. Go through the changes.

Your inconsistency in formatting and syntax makes reviewing cringe worthy.

Have you tested any of this actually works in a project?

lionslair committed:
fix more phpstan

lionslair committed:
Fix PHP Syntax Styling

Gavindecode committed:
Refactor controllers and improve file handling logic

Updated constructor syntax and refactored Search/Folders API for better readability and input handling. Improved filename generation and fixed string handling inconsistencies. Enhanced UI with better text handling in FileCell component.

Gavindecode committed:
Fix PHP Syntax Styling

Gavindecode committed:
Refactor SearchController to extend BaseController

Updated SearchController to extend BaseController instead of a custom Controller. Removed unnecessary fallback logic for the `$parent` variable, simplifying the code and improving readability.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

# Conflicts:
#	src/Http/Controllers/Api/SearchController.php

Gavindecode committed:
Ignore static analysis warning for newFileFromS3 call

Added a phpstan ignore directive to bypass a false-positive warning in the newFileFromS3 method. This prevents unnecessary linting issues during static analysis.

Gavindecode committed:
Update `package-lock.json` to reflect dependency changes

Upgraded dependencies and updated lockfile to version 3 in `package-lock.json`. The changes include updates to existing packages and addition of new devDependencies, aligning with the latest package requirements and improving compatibility and security.

lionslair committed:
ran npm install

Gavindecode committed:
Remove `package-lock.json` from the repository

The `package-lock.json` file has been deleted to prevent it from being tracked in the repository. This change may have been made to resolve dependency conflicts, enforce dynamic versioning, or to standardize dependency management strategies. Ensure team workflows are updated to align with this decision.

Gavindecode committed:
Merge branch 'feature/ELA-51-general-asset-upload-repository-public' of github.com:DCODE-GROUP/fileman into feature/ELA-51-general-asset-upload-repository-public

* 'feature/ELA-51-general-asset-upload-repository-public' of github.com:DCODE-GROUP/fileman:
  ran npm install

Gavindecode committed:
Fix PHP Syntax Styling

Gavindecode committed:
Refactor route definitions into organized classes.

Grouped web and API route definitions into dedicated classes for better structure and maintainability. Updated both Web.php and Api.php to encapsulate routes within static methods using `Route::group`.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

# Conflicts:
#	src/Routes/Api.php
#	src/Routes/Web.php

Gavindecode committed:
Remove route registration from service provider

The route registration logic was removed from the `FileManServiceProvider` and is now handled elsewhere. This improves separation of concerns, making the service provider simpler and more focused.

Gavindecode committed:
Refactor route definitions for configurability

Updated routes throughout the application to dynamically use configurable route names and paths from the `fileman` configuration file. This enhances flexibility and allows customization of route prefixes and names without altering the core codebase.

Gavindecode committed:
Fix PHP Syntax Styling

Gavindecode committed:
Add environment-based config options for fileman routing

Introduced new config options to define route and API paths/names using environment variables. Added PHPStan ignore comments to suppress static analysis warnings for these changes. This improves flexibility in configuring fileman routes.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

Gavindecode committed:
Add "view file" feature and improve file detail handling

Introduced a "view file" action with an associated eye icon and updated the file detail functionality to improve callback handling. Removed unnecessary `console.log` statements across components for cleaner code and optimized variable usage in `FileDetail.vue`.

Gavindecode committed:
Fix PHP Syntax Styling

Gavindecode committed:
Update readme with Tailwind config and Fileman setup instructions

Clarified the Tailwind content configuration block and added instructions for registering Fileman in app.js. These updates improve integration guidance for Tailwind and Fileman with the application.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

andrewsirianni commented on config/fileman.php:
```diff
@@ -1,5 +1,393 @@
 <?php
 
 return [
```
Should we consider limiting this?  I can't see we'll need this ... and may cause problems allowing some of these?

Gavindecode replied:
these files are to categorised the files based on their type. I will limit some of the files based on the mime types

andrewsirianni commented on config/fileman.php:
```diff
@@ -1,5 +1,393 @@
+            'application/json',
+            'text/csv',
+        ],
+        'Executables' => [
```
Definitely not these ...

andrewsirianni commented on config/fileman.php:
```diff
@@ -1,5 +1,393 @@
+            'application/octet-stream',
+            'application/x-sh',
+        ],
+        'Scripts' => [
```
Definitely not these ...

andrewsirianni commented on config/fileman.php:
```diff
@@ -1,5 +1,393 @@
+            'application/font-woff2',
+            'application/vnd.ms-fontobject',
+        ],
+        'Frontend' => [
```
Definitely not these ...

andrewsirianni commented on package.json:
```diff
@@ -4,14 +4,35 @@
     },
     "devDependencies": {
+        "@babel/eslint-parser": "^7.23.0",
         "@fortawesome/fontawesome-free": "^5.11.2",
```
We should remove fontawesome - change to use the Hero icons we're using.

andrewsirianni reviewed (commented):
2 considerations:

- File names - need to be SEO considerate
- Signed URLs - will we need this?

Gavindecode committed:
Add file type validation to restrict disallowed formats

Implemented a validation rule to block restricted file types based on MIME types defined in the configuration. Introduced `ValidFileType` rule and updated `FileService` and `FileRequest` to enforce this validation. Also added a corresponding error message for invalid file types.

Gavindecode committed:
Fix PHP Syntax Styling

Gavindecode committed:
Remove FontAwesome dependencies from devDependencies

FontAwesome-related packages were removed from the devDependencies section of the package.json file. These were likely unused or no longer needed for the project's development. This helps reduce unnecessary package clutter and potential maintenance overhead.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

Gavindecode committed:
Fix eslint issues

Gavindecode committed:
Refactor file handling to simplify URL and naming logic.

Removed preview-related methods and logic from the File model and resources, defaulting to the main URL. Introduced `getSeoFriendlyName` for generating clean, SEO-friendly filenames. These changes streamline file handling, improve naming consistency, and reduce unnecessary complexity.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

Gavindecode committed:
Fix PHP Syntax Styling

Gavindecode committed:
Remove unused getImageMimes method from File model

The getImageMimes method was unused and has been removed for clarity and maintainability. Cleaning up unused code helps reduce potential confusion and keeps the codebase lean.

Gavindecode committed:
Merge remote-tracking branch 'origin/feature/ELA-51-general-asset-upload-repository-public' into feature/ELA-51-general-asset-upload-repository-public

Gavindecode commented:
> 2 considerations:
> 
> * File names - need to be SEO considerate
> * Signed URLs - will we need this?

Updated, thanks for the feedbacks.

Gavindecode committed:
Fix PHP Syntax Styling
